File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int n = nums.size();
        int count = 0;

        for(int i=1; i<n ; i++) {
            if(nums[i] <= nums[i-1]) {
                int newValue = nums[i-1] + 1;

                count += newValue - nums[i];
                nums[i] = newValue;
            }
        }
        return count;
    }
};

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.py
class Solution(object):
    def minIncrementForUnique(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()

        n = len(nums)
        count = 0 

        for i in range(1,n) :
            if(nums[i] <= nums[i-1]) :
                newValue = nums[i-1] + 1
                count += newValue - nums[i]
                nums[i] = newValue
        return count